aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1132 @@
+    /// address is actually a shareholder address, just return the address back.
+    ///
+    /// This returns 0x0 if no shareholder is found for the given beneficiary / the address is not a shareholder itself.
+    public fun shareholder(vesting_contract_address: address, shareholder_or_beneficiary: address): address acquires VestingContract {
+        assert_active_vesting_contract(vesting_contract_address);
+
+        let shareholders = &shareholders(vesting_contract_address);
+        if (vector::contains(shareholders, &shareholder_or_beneficiary)) {
+            return shareholder_or_beneficiary
+        };
+        let vesting_contract = borrow_global<VestingContract>(vesting_contract_address);
+        let result = @0x0;
+        vector::any(shareholders, |shareholder| {
+            if (shareholder_or_beneficiary == get_beneficiary(vesting_contract, *shareholder)) {
+                result = *shareholder;
+                true
+            } else {
+                false
+            }
+        });
+
+        result
+    }
+
+    /// Create a vesting schedule with the given schedule of distributions, a vesting start time and period duration.
+    public fun create_vesting_schedule(
+        schedule: vector<FixedPoint32>,
+        start_timestamp_secs: u64,
+        period_duration: u64,
+    ): VestingSchedule {
+        assert!(vector::length(&schedule) > 0, error::invalid_argument(EEMPTY_VESTING_SCHEDULE));
+        assert!(period_duration > 0, error::invalid_argument(EZERO_VESTING_SCHEDULE_PERIOD));
+        assert!(
+            start_timestamp_secs >= timestamp::now_seconds(),
+            error::invalid_argument(EVESTING_START_TOO_SOON),
+        );
+
+        VestingSchedule {
+            schedule,
+            start_timestamp_secs,
+            period_duration,
+            last_vested_period: 0,
+        }
+    }
+
+    /// Create a vesting contract with a given configurations.
+    public fun create_vesting_contract(
+        admin: &signer,
+        buy_ins: SimpleMap<address, Coin<AptosCoin>>,
+        vesting_schedule: VestingSchedule,
+        withdrawal_address: address,
+        contract_creation_seed: vector<u8>,
+    ): address acquires AdminStore {
+        assert!(
+            !system_addresses::is_reserved_address(withdrawal_address),
+            error::invalid_argument(EINVALID_WITHDRAWAL_ADDRESS),
+        );
+        assert_account_is_registered_for_apt(withdrawal_address);
+        let shareholders_address = &simple_map::keys(&buy_ins);
+        assert!(vector::length(shareholders_address) > 0, error::invalid_argument(ENO_SHAREHOLDERS));
+
+        let shareholders = simple_map::create<address, Amount>();
+        let grant = coin::zero<AptosCoin>();
+        let grant_amount = 0;
+        vector::for_each_ref(shareholders_address, |shareholder| {
+            let shareholder: address = *shareholder;
+            let (_, buy_in) = simple_map::remove(&mut buy_ins, &shareholder);
```

Comment:
Here also, it's best to use `simple_map::to_vec_pair` to get two vectors with same index referring to `key` and its corresponding `value` in these vectors. `remove` is O(n) but you can use `vector::pop` on both the vectors iteratively.